### DISCUSSION TITLE
seek advice to localize the network  health check in mqtt_as.py

### DISCUSSION THREAD

**Discussion by jfleon** [micropython/micropython#9782] [Using MicroPython]
Hi in this library the network connection is checked in the file mqtt_py with the function wan_ok() that sent a packet to google dns  on port 53 and check its answer
I have mqtt sensors newtork that are not connected to internet only to a lan.
I am seeking advice on how to write a lan_ok function that will work in a similar way. The machine always available are the lan router  and the mqtt broker. the mqtt broker is a raspi so I guess I could install a dns server on it but I wonder if there are better options to solve this problem

Thanks

**Comment by beetlegigg** [comment]
I think you will find the wan_ok() in the mqtt_as.py  can be called if it is desired to check on an internet connected broker, but its not used if your broker is a rpi on your local network (as is mine).  For the network and broker connection checks then have a look at the _keep_connected(self).   The very premiss of the mqtt_as.py library is to be a resilient  connection that can handle wifi outages when used on the limits of wifi range and the reception can be a bit flaky and to handle re-connects automatically.  You will not need to have an additional lan_ok method as its build into the MQTTClient class in the mqtt_as.py library.  Note the co-routine you pass into the config dictionary of 'wifi_coro':     eliza,  probably to an async function  such as wifi_han that utilises an LED as an indication of the wifi state.

**Comment by peterhinch** [comment]
@beetlegigg is correct. When using `mqtt_as` on a LAN, WiFi outages are handled automatically. You can call `isconnected` at any time to establish whether local connectivity exists. If connectivity is OK you can use `broker_up` to check on broker status. If you need something beyond this, please explain your requirement in more detail.

**Comment by peterhinch** [comment]
If your query is a _general_ one, rather than specific to `mqtt_as`, the way to check connectivity on a network is to identify a resource on the net that should always be available, and to provoke it to respond. Hence my use of a DNS query against `8.8.8.8`. On a local network I would probably send a ping packet to the router and check for a response. There is a MicroPython ping application: search the old forum if you want to use it.
